nav-app/src/app/datatable/data-table.component.ts

```diff
@@ -285,13 +285,26 @@ export class DataTableComponent implements AfterViewInit {
     /**
      * Show all sub-techniques in layout view
      */
-    expandSubtechniques(): void {
+    expandSubtechniques(showAnnotatedOnly?: boolean): void {
         if (this.viewModel.layout.layout == "mini") return; //control disabled in mini layout
         for (let technique of this.dataService.getDomain(this.viewModel.domainID).techniques) {
             if (technique.subtechniques.length > 0) {
                 for (let id of technique.get_all_technique_tactic_ids()) {
                     let tvm = this.viewModel.getTechniqueVM_id(id);
-                    tvm.showSubtechniques = true;
+                    if (!showAnnotatedOnly) {
+                        tvm.showSubtechniques = true;
+                    } else {
+                        let foundAnnotated = false;
+                        for (let subtechnique of technique.subtechniques) {
+                            for (let sid of subtechnique.get_all_technique_tactic_ids()) {
+                                let svm = this.viewModel.getTechniqueVM_id(sid);
+                                if (svm.annotated() && !foundAnnotated) {
+                                    foundAnnotated = true; // set boolean after it finds the first annotated subtechnique
```

Comment:
I believe you can `break` here for some added efficiency. Either that or collapse the whole thing into `subtechnique.get_all_technique_tactic_ids().some((sid) => { ... })` [which probably does the same thing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).